src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1219 @@
+            sage: print(S)
+            A seed with cluster variables [x1, x2, x3] and exchange polynomials [x2*a2 + x3*a3 + a0, x1*b1 + x3*b3 + b0, x1*c1 + x2*c2 + c0]
+
+        More complicated polynomials are allowed, as long as they are
+        irreducible::
+
+            sage: var('x1,x2,x3')
+            (x1, x2, x3)
+            sage: S = LpaSeed({x1: 1 + x2*x3^2 + 4*x3^3, x2: 2 - x1^2, x3: 4 + x1^3*x2^2 - 3*x1})
+            sage: print(S)
+            A seed with cluster variables [x1, x2, x3] and exchange polynomials [x2*x3^2 + 4*x3^3 + 1, -x1^2 + 2, x1^3*x2^2 - 3*x1 + 4]
+
+        Nonirreducible polynomials will raise an exception::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: S = LpaSeed({x1: 4 - x2^2, x2: 1 + x1})
+            Traceback (most recent call last):
+            ...
+            ValueError: (LP2) fail: -x2^2 + 4 is not irreducible over Integer Ring
+
+        Different base rings are allowed::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LpaSeed({x1: 1 + x2, x2: 1 + x1}, base_ring=QQ)
+            sage: print(S)
+            A seed with cluster variables [x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+        """
+
+        self._names = None  # the symbolic variables we use for our polynomials
+        self._base_ring = None  # the coefficient ring's base ring, A
+        self._coefficients = None  # coefficients we join to A to get ring
+        self._ambient_field = None  # function field F in `_names` over Frac(A)
+        self._polynomial_ring = None  # exchange polynomials parent ring
+        self._rank = None  # transcendence degree of F over Frac(A)
+        self._exchange_polys = None  # tuple of polynomials in `_names`
+        self._laurent_polys = None  # tuple of laurent polynomials in `_names`
+        self._cluster_vars = None  # cluster variables for this seed
+        self._mutation_sequence = None  # sequence we have already mutated at
+
+        # unpack supplied data
+
+        # make a copy
+        if isinstance(data, LpaSeed):
+
+            self._names = data._names
+            self._base_ring = data._base_ring
+            self._coefficients = data._coefficients
+            self._ambient_field = data._ambient_field
+            self._polynomial_ring = data._polynomial_ring
+            self._rank = data._rank
+            self._exchange_polys = copy(data._exchange_polys)
+            self._laurent_polys = copy(data._laurent_polys)
+            self._cluster_vars = copy(data._cluster_vars)
+            self._mutation_sequence = copy(data._mutation_sequence)
+
+        # we assume we have dictionary of variable/poly pairs
+        elif isinstance(data, dict):
+
+            names = tuple(data.keys())  # ensure names are immutable
+
+            # if we are not supplied any coefficients, check input data
+            if coefficients == ():
```

Comment:
```suggestion
            if not coefficients:
```
This is faster and also handles lists and other similar things that behave like a tuple.